internal/scan-manager/scanners/veracode/src/main/java/org/wso2/security/tools/scanmanager/scanners/veracode/service/VeracodeScanner.java

```diff
@@ -0,0 +1,304 @@
+        options._vid = VeracodeScannerConfiguration.getInstance().getApiId();
+        options._vkey = String.valueOf(VeracodeScannerConfiguration.getInstance().getApiKey());
+        createLogWriter(options);
+        getUploadAPIWrapper(options);
+        getResultAPIWrapper(options);
+    }
+
+    /**
+     * Initialising the Veracode configurations.
+     *
+     * @throws IOException
+     */
+    private static void loadConfiguration() throws IOException {
+        if (YAMLConfigurationReader.getInstance().getConfigs() == null) {
+            YAMLConfigurationReader.getInstance().loadConfiguration();
+        }
+
+        VeracodeScannerConfiguration.getInstance().setApiId(YAMLConfigurationReader.getInstance()
+                .getConfigProperty(VeracodeScannerConstants.VERACODE_API_ID));
+        VeracodeScannerConfiguration.getInstance().setApiKey((YAMLConfigurationReader.getInstance()
+                .getConfigProperty(VeracodeScannerConstants.VERACODE_API_KEY)).toCharArray());
+        VeracodeScannerConfiguration.getInstance().setOutputFolderPath(YAMLConfigurationReader.getInstance()
+                .getConfigProperty(VeracodeScannerConstants.VERACODE_OUTPUT_FOLDER_PATH));
+        VeracodeScannerConfiguration.getInstance().setOutputFilePath(YAMLConfigurationReader.getInstance()
+                .getConfigProperty(VeracodeScannerConstants.VERACODE_OUTPUT_FOLDER_PATH) + File.separator +
+                YAMLConfigurationReader.getInstance().getConfigProperty(VeracodeScannerConstants
+                        .VERACODE_OUTPUT_FILE_NAME));
+        VeracodeScannerConfiguration.getInstance().setLogFilePath(YAMLConfigurationReader.getInstance().
+                getConfigProperty(VeracodeScannerConstants.VERACODE_LOG_FILE_PATH));
+        VeracodeScannerConfiguration.getInstance().setScannerClass(YAMLConfigurationReader.getInstance().
+                getConfigProperty(VeracodeScannerConstants.SCANNER_BEAN_CLASS_NAME));
+    }
+
+    /**
+     * Run the scan using product zip file.
+     *
+     * @param scanRequest Object that represent the required information for the scanner operation
+     * @return ResponseEntity with status of the request
+     */
+    @Override
+    public ResponseEntity startScan(ScannerScanRequest scanRequest) {
+        scanContext = new ScanContext();
+        scanContext.setJobId(scanRequest.getJobId());
+        scanContext.setAppId(scanRequest.getAppId());
+        scanContext.setArtifactLocation(scanRequest.getFileMap().get(VeracodeScannerConstants.SCAN_ARTIFACT).get(0));
+        ResponseEntity responseEntity;
+
+        if (scanRequest.getFileMap().get(VeracodeScannerConstants.SCAN_ARTIFACT) != null) {
+            if (!StringUtils.isEmpty(scanContext.getAppId())) {
+                responseEntity = new ResponseEntity<>(HttpStatus.ACCEPTED);
+                startVeracodeScan();
+            } else {
+                String message = "Error occured while submitting the start scan request since " +
```

Comment:
Same error logic repeated in two locations (next else). Can we have a `Util` method to do the repetitive task here and call it with `message` and `status`?

Reply:
Same comment applies to all other service methods. Please check other locations too.